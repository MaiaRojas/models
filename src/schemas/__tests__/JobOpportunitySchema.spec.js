const mongoose = require('mongoose/browser');
const { OrganizationMembershipSchema, JobOpportunitySchema, CampusSchema } = require('../')(mongoose);

describe('Opportunity', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, JobOpportunitySchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate ...', (done) => {
    const employer = new mongoose.Document({}, OrganizationMembershipSchema);
    const campus = new mongoose.Document({}, CampusSchema);

    const opportunity = new mongoose.Document({
      employer: employer._id,
      address: ' Av. Almirante Grau',
      city: 'Lima',
      campus: campus._id,
      vacancies: 3,
      title: 'Desenvolvedor Front-end Junior',
      description: 'Mussum Ipsum, cacilds vidis litro abertis. Per aumento de cachacis',
      type: 'Front-end Developer',
      hireType: 'payroll',
      scheduleType: 'part-time',
      salaryRange: 'R$3.501 to R$4.000',
      tentativeStart: new Date(),
      active: true,
    }, JobOpportunitySchema);

    opportunity.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
