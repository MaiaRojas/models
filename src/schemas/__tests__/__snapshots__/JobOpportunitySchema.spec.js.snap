// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Opportunity should fail validation when fields missing 1`] = `
Object {
  "active": [ValidatorError: Path \`active\` is required.],
  "address": [ValidatorError: Path \`address\` is required.],
  "campus": [ValidatorError: Path \`campus\` is required.],
  "city": [ValidatorError: Path \`city\` is required.],
  "description": [ValidatorError: Path \`description\` is required.],
  "employer": [ValidatorError: Path \`employer\` is required.],
  "salaryRange": [ValidatorError: Path \`salaryRange\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "vacancies": [ValidatorError: Path \`vacancies\` is required.],
}
`;
